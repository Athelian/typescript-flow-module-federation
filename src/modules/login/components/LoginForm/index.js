// @flow
import * as React from 'react';
import * as Yup from 'yup';
import { FormattedMessage } from 'react-intl';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import faSignInAlt from '@fortawesome/fontawesome-pro-solid/faSignInAlt';
import messages from 'modules/login/messages';
import { LoginBoxStyle } from 'modules/login/style';
import { Form, Field } from 'components/Form';
import TextInput from 'components/Form/TextInput';
import { CustomButton } from 'components/NavButtons';
import yupToFormErrors from 'utils/yupToFormErrors';

type Props = {
  onLogin: Function,
};

const LoginSchema = Yup.object().shape({
  email: Yup.string()
    .required(<FormattedMessage {...messages.required} />)
    .email(<FormattedMessage {...messages.emailError} />),
  password: Yup.string().required(<FormattedMessage {...messages.required} />),
});

const onValidate = (values: Object) =>
  new Promise((resolve, reject) => {
    LoginSchema.validate(values, { abortEarly: false })
      .then(() => resolve({}))
      .catch(error => reject(yupToFormErrors(error)));
  });

function LoginForm({ onLogin }: Props) {
  return (
    <Form
      initialValues={{
        email: '',
        password: '',
      }}
      validateOnChange
      validations={onValidate}
      onSubmit={onLogin}
      render={({ errors, touched, handleSubmit, isInvalid, isDirty, setFieldValue }) => (
        <form data-testid="loginForm" onSubmit={handleSubmit}>
          <div className={LoginBoxStyle}>
            <Field
              name="email"
              render={({ input }) => (
                <TextInput
                  {...input}
                  data-testid="email"
                  type="email"
                  title={<FormattedMessage {...messages.email} />}
                  error={touched.email && errors.email}
                  vertical
                  forceHoverStyle
                  onChange={setFieldValue}
                />
              )}
            />
            <Field
              name="password"
              render={({ input }) => (
                <TextInput
                  {...input}
                  data-testid="password"
                  type="password"
                  title={<FormattedMessage {...messages.password} />}
                  error={touched.password && errors.password}
                  vertical
                  forceHoverStyle
                  onChange={setFieldValue}
                />
              )}
            />
            <CustomButton
              data-testid="submitButton"
              label={<FormattedMessage {...messages.login} />}
              icon={<FontAwesomeIcon icon={faSignInAlt} fixedWidth />}
              color="teal"
              type="submit"
              disabled={isInvalid || !isDirty}
              style={{ boxShadow: 'none', marginTop: '20px' }}
            />
          </div>
        </form>
      )}
    />
  );
}

export default LoginForm;
