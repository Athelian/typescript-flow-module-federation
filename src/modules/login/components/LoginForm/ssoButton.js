// @flow
import * as React from 'react';
import { SsoButtonStyle, LogoStyle } from 'modules/login/style';
import googleIcon from 'media/google.png';

type Props = {|
  type: string,
|};

const SsoButton = ({ type }: Props) => {
  const serverURL = process.env.ZENPORT_SERVER_URL || '';
  return (
    <a href={`${serverURL}/sso/login/${type}`} style={{ width: '100%' }}>
      <button type="button" className={SsoButtonStyle}>
        {type === 'google' && (
          <>
            <img src={googleIcon} className={LogoStyle} alt="google logo" />
            Continue with Google
          </>
        )}
      </button>
    </a>
  );
};

export default SsoButton;
