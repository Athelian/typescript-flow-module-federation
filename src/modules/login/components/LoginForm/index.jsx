// @flow
import * as React from 'react';
import { Form, Field } from 'zenform';
import * as Yup from 'yup';
import { FormattedMessage } from 'react-intl';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import faSignInAlt from '@fortawesome/fontawesome-pro-solid/faSignInAlt';
import messages from 'modules/login/messages';
import { LoginBoxStyle } from 'modules/login/style';
import TextInput from 'components/TextInput';
import { CustomButton } from 'components/NavButtons';
import yupToFormErrors from 'utils/yupToFormErrors';

type Props = {
  onLogin: Function,
};

const LoginSchema = Yup.object().shape({
  email: Yup.string()
    .required(<FormattedMessage {...messages.required} />)
    .email(<FormattedMessage {...messages.emailError} />),
  password: Yup.string().required(<FormattedMessage {...messages.required} />),
});

const onValidate = (values: Object) =>
  new Promise((resolve, reject) => {
    LoginSchema.validate(values, { abortEarly: false })
      .then(() => resolve({}))
      .catch(error => reject(yupToFormErrors(error)));
  });

function LoginForm({ onLogin }: Props) {
  return (
    <Form
      initialValues={{
        email: '',
        password: '',
      }}
      validateOnChange
      validations={onValidate}
      onSubmit={onLogin}
      render={({ errors, touched, onSubmit, isInvalid }) => (
        <form data-testid="loginForm" onSubmit={onSubmit}>
          <div className={LoginBoxStyle}>
            <Field
              name="email"
              render={({ input }) => (
                <TextInput
                  {...input}
                  data-testid="email"
                  type="email"
                  title={<FormattedMessage {...messages.email} />}
                  error={touched.email && errors.email}
                />
              )}
            />
            <Field
              name="password"
              render={({ input }) => (
                <TextInput
                  {...input}
                  data-testid="password"
                  type="password"
                  title={<FormattedMessage {...messages.password} />}
                  error={touched.password && errors.password}
                />
              )}
            />
            <CustomButton
              data-testid="submitButton"
              label={<FormattedMessage {...messages.login} />}
              icon={<FontAwesomeIcon icon={faSignInAlt} fixedWidth />}
              color="teal"
              type="submit"
              disabled={isInvalid}
              style={{ boxShadow: 'none', marginTop: '20px' }}
            />
          </div>
        </form>
      )}
    />
  );
}

export default LoginForm;
