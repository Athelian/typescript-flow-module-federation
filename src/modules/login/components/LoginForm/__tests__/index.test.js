import React from 'react';
import { Provider } from 'unstated';
import '@testing-library/jest-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import LanguageProvider from 'modules/language';
import LoginForm from '../index';

afterEach(cleanup);

describe('<LoginForm />', () => {
  it('should render without crash', () => {
    const onLoginFn = jest.fn();
    const { getByTestId, container } = render(
      <Provider>
        <LanguageProvider>
          <LoginForm onLogin={onLoginFn} />
        </LanguageProvider>
      </Provider>
    );
    const emailElement = getByTestId('email');
    const passwordElement = getByTestId('password');
    const submitButtonElement = getByTestId('submitButton');

    expect(container).toMatchSnapshot();
    expect(emailElement).not.toBeNull();
    expect(passwordElement).not.toBeNull();
    expect(submitButtonElement).not.toBeNull();
    expect(submitButtonElement.hasAttribute('disabled')).toBeTruthy();
  });

  it('should allow to click on submit button on valid data', () => {
    const onLoginFn = jest.fn();
    const { getByTestId, container } = render(
      <Provider>
        <LanguageProvider>
          <LoginForm onLogin={onLoginFn} />
        </LanguageProvider>
      </Provider>
    );
    const emailElement = getByTestId('email');
    const passwordElement = getByTestId('password');
    const submitButtonElement = getByTestId('submitButton');

    expect(container).toMatchSnapshot();
    emailElement.value = 'test@example.com';
    passwordElement.value = 'demo';
    fireEvent.change(emailElement);
    fireEvent.change(passwordElement);
    fireEvent.click(submitButtonElement);
    expect(container).toMatchSnapshot();
  });
});
