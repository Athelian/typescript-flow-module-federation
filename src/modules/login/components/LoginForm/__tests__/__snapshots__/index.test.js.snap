// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should allow to click on submit button on valid data 1`] = `
.emotion-12 {
  box-shadow: 0 10px 30px 0 rgba(0,0,0,0.1);
  border-radius: 5px;
  background-color: #fff;
  padding: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: grid;
  grid-auto-rows: min-content;
  grid-gap: 40px;
  justify-items: center;
}

.emotion-0 {
  position: absolute;
  left: -20px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 0;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  color: #aaa;
  background-color: #ccc;
  border-radius: 999px;
  height: 30px;
  padding: 0 10px;
  cursor: default;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: 75px;
}

.emotion-11:focus {
  outline: 0;
}

.emotion-9 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 12px;
}

.emotion-10 {
  margin: 0 0 0 5px;
  font-size: 12px;
}

.emotion-8 {
  display: grid;
  grid-auto-flow: row;
  grid-auto-rows: min-content;
  grid-gap: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-1 {
  width: 100%;
  font-size: 14px;
  min-height: 20px;
  line-height: 20px;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #aaa;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0 5px;
  text-align: left;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  border-radius: 5px;
  background-color: #ffffff;
  height: 30px;
  width: 200px;
  min-width: 200px;
  cursor: text;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  box-shadow: 0 3px 10px rgba(0,0,0,0.2);
}

.emotion-2 > button {
  opacity: 1;
}

.emotion-2 > input {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-2 > input,
.emotion-2 > textarea {
  border: none;
  padding: 0;
  width: 100%;
  height: 100%;
  line-height: 20px;
  padding: 0 5px;
  background: none;
  border-radius: 5px;
  color: #555555;
  font-size: 14px;
  font-weight: bold;
}

.emotion-2 > input:focus,
.emotion-2 > textarea:focus {
  outline: none;
}

.emotion-2 > input::-webkit-input-placeholder,
.emotion-2 > textarea::-webkit-input-placeholder {
  color: #ccc;
}

.emotion-2 > input::-moz-placeholder,
.emotion-2 > textarea::-moz-placeholder {
  color: #ccc;
}

.emotion-2 > input:-ms-input-placeholder,
.emotion-2 > textarea:-ms-input-placeholder {
  color: #ccc;
}

.emotion-2 > input::placeholder,
.emotion-2 > textarea::placeholder {
  color: #ccc;
}

<div>
  <form
    data-testid="loginForm"
  >
    <div
      class="emotion-12"
    >
      <div
        class="emotion-8"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-0"
          />
          <div
            class="emotion-1"
          >
            EMAIL
          </div>
          <div
            class="emotion-2"
          >
            <input
              data-testid="email"
              name="email"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="emotion-3"
        >
          <div
            class="emotion-0"
          />
          <div
            class="emotion-1"
          >
            PASSWORD
          </div>
          <div
            class="emotion-2"
          >
            <input
              data-testid="password"
              name="password"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="emotion-11"
        data-testid="submitButton"
        disabled=""
        type="submit"
      >
        <div
          class="emotion-9"
        >
          LOGIN
        </div>
        <div
          class="emotion-10"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-in-alt fa-w-16 fa-fw "
            data-icon="sign-in-alt"
            data-prefix="fas"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should allow to click on submit button on valid data 2`] = `
<div>
  <form
    data-testid="loginForm"
  >
    <div
      class="css-13fmzcs"
    >
      <div
        class="css-qwdb27"
      >
        <div
          class="css-p52le"
        >
          <div
            class="css-1n42b97"
          />
          <div
            class="css-1o4txpy"
          >
            EMAIL
          </div>
          <div
            class="css-1b8yj6f"
          >
            <input
              data-testid="email"
              name="email"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="css-p52le"
        >
          <div
            class="css-1n42b97"
          />
          <div
            class="css-1o4txpy"
          >
            PASSWORD
          </div>
          <div
            class="css-1b8yj6f"
          >
            <input
              data-testid="password"
              name="password"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="css-198dvuf"
        data-testid="submitButton"
        disabled=""
        type="submit"
      >
        <div
          class="css-6tmki"
        >
          LOGIN
        </div>
        <div
          class="css-x1sjjo"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-in-alt fa-w-16 fa-fw "
            data-icon="sign-in-alt"
            data-prefix="fas"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should render without crash 1`] = `
.emotion-12 {
  box-shadow: 0 10px 30px 0 rgba(0,0,0,0.1);
  border-radius: 5px;
  background-color: #fff;
  padding: 20px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: grid;
  grid-auto-rows: min-content;
  grid-gap: 40px;
  justify-items: center;
}

.emotion-0 {
  position: absolute;
  left: -20px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 0;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  color: #aaa;
  background-color: #ccc;
  border-radius: 999px;
  height: 30px;
  padding: 0 10px;
  cursor: default;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: 75px;
}

.emotion-11:focus {
  outline: 0;
}

.emotion-9 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 12px;
}

.emotion-10 {
  margin: 0 0 0 5px;
  font-size: 12px;
}

.emotion-8 {
  display: grid;
  grid-auto-flow: row;
  grid-auto-rows: min-content;
  grid-gap: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-1 {
  width: 100%;
  font-size: 14px;
  min-height: 20px;
  line-height: 20px;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #aaa;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0 5px;
  text-align: left;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  border-radius: 5px;
  background-color: #ffffff;
  height: 30px;
  width: 200px;
  min-width: 200px;
  cursor: text;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  box-shadow: 0 3px 10px rgba(0,0,0,0.2);
}

.emotion-2 > button {
  opacity: 1;
}

.emotion-2 > input {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-2 > input,
.emotion-2 > textarea {
  border: none;
  padding: 0;
  width: 100%;
  height: 100%;
  line-height: 20px;
  padding: 0 5px;
  background: none;
  border-radius: 5px;
  color: #555555;
  font-size: 14px;
  font-weight: bold;
}

.emotion-2 > input:focus,
.emotion-2 > textarea:focus {
  outline: none;
}

.emotion-2 > input::-webkit-input-placeholder,
.emotion-2 > textarea::-webkit-input-placeholder {
  color: #ccc;
}

.emotion-2 > input::-moz-placeholder,
.emotion-2 > textarea::-moz-placeholder {
  color: #ccc;
}

.emotion-2 > input:-ms-input-placeholder,
.emotion-2 > textarea:-ms-input-placeholder {
  color: #ccc;
}

.emotion-2 > input::placeholder,
.emotion-2 > textarea::placeholder {
  color: #ccc;
}

<div>
  <form
    data-testid="loginForm"
  >
    <div
      class="emotion-12"
    >
      <div
        class="emotion-8"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-0"
          />
          <div
            class="emotion-1"
          >
            EMAIL
          </div>
          <div
            class="emotion-2"
          >
            <input
              data-testid="email"
              name="email"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="emotion-3"
        >
          <div
            class="emotion-0"
          />
          <div
            class="emotion-1"
          >
            PASSWORD
          </div>
          <div
            class="emotion-2"
          >
            <input
              data-testid="password"
              name="password"
              placeholder=""
              spellcheck="false"
              style="text-align: left;"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="emotion-11"
        data-testid="submitButton"
        disabled=""
        type="submit"
      >
        <div
          class="emotion-9"
        >
          LOGIN
        </div>
        <div
          class="emotion-10"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-in-alt fa-w-16 fa-fw "
            data-icon="sign-in-alt"
            data-prefix="fas"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </form>
</div>
`;
